src/components/PostDetails.tsx

```diff
@@ -1,106 +1,154 @@
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [currentError, setCurrentError] = useState<Errors | null>(null);
+
+  const handleDeleteComment = (comment: Comment) => {
+    setCurrentError(null);
+    setComments(comments.filter(c => c.id !== comment.id));
+
+    const restoreDeletedComment = () => {
+      setComments(prevComments => [...prevComments, comment]);
+      setCurrentError(Errors.CommentDeleting);
+    };
+
+    deleteComment(comment.id)
+      .then((response: unknown) => {
+        if (response && typeof response === 'object' && 'error' in response) {
+          restoreDeletedComment();
+        }
+      })
+      .catch(() => {
+        restoreDeletedComment();
+      });
+  };
+
+  useEffect(() => {
+    if (!post) {
+      return;
+    }
+
+    setCurrentError(null);
+    setIsCommentsLoading(true);
+    setComments([]);
+
+    getComments(id)
+      .then(response => {
+        if (Array.isArray(response) && response.length > 0) {
+          setComments(response);
+        }
+
+        if (!Array.isArray(response)) {
+          setCurrentError(Errors.CommentsLoading);
+        }
+
+        if (Array.isArray(response) && response.length === 0) {
+          setCurrentError(Errors.NoComments);
+        }
+      })
+      .catch(() => setCurrentError(Errors.CommentsLoading))
+      .finally(() => {
+        setIsCommentsLoading(false);
+      });
+  }, [post]);
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
+          <h2 data-cy="PostTitle">{`#${id}: ${title}`}</h2>
+
+          <p data-cy="PostBody">{body}</p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+          {isCommentsLoading && <Loader />}
```

Comment:
Ensure that the loader and error messages are not displayed simultaneously. Consider updating the conditions to make these states mutually exclusive.